easybuild/framework/easyconfig/types.py

```diff
@@ -301,23 +301,31 @@ def to_dependency(dep):
     #   (name, version[, versionsuffix[, toolchain]])
 
     if isinstance(dep, dict):
-        depspec = {}
-        found_name_version = False
-        for key, value in dep.items():
-            if key in ['name', 'version', 'versionsuffix']:
-                depspec[key] = value
-            elif key == 'toolchain':
-                depspec['toolchain'] = to_name_version_dict(value)
-            elif not found_name_version:
-                depspec.update({'name': key, 'version': value})
+        if 'external' in dep and dep['external']:
+            if len(dep) == 2 and 'name' in dep:
+                depspec = dep
```

Comment:
ah, yeah, it should be there, `_parse_dependency` picks up on it

then make sure it's always there